src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -1,38 +1,130 @@
 package org.constellation.domain.redownload
 
+import cats.effect.{ContextShift, LiftIO, Sync}
 import cats.effect.Concurrent
 import cats.effect.concurrent.Ref
 import cats.implicits._
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import org.constellation.DAO
+import org.constellation.consensus.EdgeProcessor
+import org.constellation.domain.snapshotInfo.SnapshotInfoChunk
 import org.constellation.schema.Id
+import org.constellation.storage.RecentSnapshot
+import org.constellation.util.MajorityStateChooser.NodeSnapshots
+import org.constellation.util.{HealthChecker, MajorityStateChooser}
 
-class RedownloadService[F[_]]()(implicit F: Concurrent[F]) {
+import scala.concurrent.duration._
+
+class RedownloadService[F[_]](dao: DAO)(implicit F: Concurrent[F], C: ContextShift[F]) {
+  val logger = Slf4jLogger.getLogger[F]
 
   // TODO: Consider Height/Hash type classes
-  private[redownload] val ownSnapshots: Ref[F, Map[Long, String]] = Ref.unsafe(Map.empty)
-  private[redownload] val peersProposals: Ref[F, Map[Long, (Id, String)]] = Ref.unsafe(Map.empty)
+  private[redownload] val ownSnapshots: Ref[F, Map[Long, RecentSnapshot]] = Ref.unsafe(Map.empty)
+  private[redownload] val peersProposals
+    : Ref[F, Map[Long, Seq[NodeSnapshots]]] = Ref.unsafe(Map.empty) //todo join/leave?
 
-  def persistOwnSnapshot(height: Long, hash: String): F[Unit] =
+  def persistOwnSnapshot(height: Long, recentSnapshot: RecentSnapshot): F[Unit] =
     ownSnapshots.modify { m =>
-      val updated = if (m.get(height).nonEmpty) m else m.updated(height, hash)
+      val updated = if (m.get(height).nonEmpty) m else m.updated(height, recentSnapshot)
       (updated, ())
     }
 
-  def getOwnSnapshots(): F[Map[Long, String]] = ownSnapshots.get
+  def getOwnSnapshots(): F[Map[Long, RecentSnapshot]] = ownSnapshots.get
 
-  def getOwnSnapshot(height: Long): F[Option[String]] =
+  def getOwnSnapshot(height: Long): F[Option[RecentSnapshot]] =
     ownSnapshots.get.map(_.get(height))
 
-  // TODO: Use cluster peers to fetch "/snapshot/own" and merge with peersProposals Map
-  def fetchPeersProposals(): F[Unit] = ???
+  def getProposals() =
+    for {
+      peers <- LiftIO[F].liftIO(dao.readyPeers)
+      chunkedPeerProposalMaps <- peers.toList.traverse {
+        case (id, peerData) =>
+          val resp = peerData.client.getNonBlockingFLogged[F, Array[Array[Byte]]](
+            "snapshot/own",
+            timeout = 45.seconds, //todo change these?
+            tag = "snapshot/obj/snapshot"
+          )(C)
+          resp.map((id, _))
+      }
+    } yield chunkedPeerProposalMaps
+
+  def updatePeerProps(fetchedProposals: Seq[(Id, RecentSnapshot)]) =
```

Comment:
Please keep the consistent naming. Either proposals or props. Currently it's mixed across one class